examples/ecs/computed_states.rs

```diff
@@ -0,0 +1,675 @@
+//! This example illustrates the use of [`ComputedStates`] for more complex state handling patterns.
+//!
+//! In this case, we'll be implementing the following pattern:
+//! - The game will start in a `Menu` state, which we can return to with `Esc`
+//! - From there, we can enter the game - where our bevy symbol moves around and changes color
+//! - While in game, we can pause and unpause the game using `Space`
+//! - We can also toggle "Turbo Mode" with the `T` key - where the movement and color changes are all faster. This
+//!   is retained between pauses, but not if we exit to the main menu.
+//!
+//! In addition, we want to enable a "tutorial" mode, which will involve it's own state that is toggled in the main menu.
+//! This will display instructions about movement and turbo mode when in game and unpaused, and instructions on how to unpause when paused.
```

Comment:
Can we define all the states and their relations here?
The `ComputeState` trait impl takes a lot of space and it may be difficult to understand the purpose when it's so spread apart so much.

Reply:
Done